packages/gestalt/src/SideNavigation/GroupContent.js

```diff
@@ -92,61 +101,93 @@ export default function SideNavigationGroupContent({
     }
   }, [hovered, focused, primaryAction, forceIconButton, isMobile, showIconButton]);
 
+  const hasBorder = sideNavigationCollapsed
+    ? hasActiveChild
+    : expanded && selectedItemId === itemId;
+
+  const collapsed = sideNavigationCollapsed && !overlayPreview;
+
   return (
     <Box
       color={itemColor ?? undefined}
-      paddingY={2}
+      paddingY={collapsed ? undefined : 2}
+      padding={collapsed ? 3 : undefined}
+      width={collapsed ? 44 : undefined}
+      height={collapsed ? 44 : undefined}
       minHeight={44}
       rounding={2}
       display="flex"
       alignItems="center"
+      position="relative"
       dangerouslySetInlineStyle={{
-        __style:
-          expanded && selectedItemId === itemId
-            ? {
-                border: `2px solid ${TOKEN_COLOR_BACKGROUND_SELECTED_BASE}`,
-                ...paddingStyle,
-              }
-            : paddingStyle,
+        __style: hasBorder
+          ? {
+              border: `2px solid ${TOKEN_COLOR_BACKGROUND_SELECTED_BASE}`,
+              ...paddingStyle,
+            }
+          : paddingStyle,
       }}
     >
+      {collapsed && icon && notificationAccessibilityLabel ? (
+        <Box
+          aria-label={notificationAccessibilityLabel}
+          height={8}
+          width={8}
+          rounding="circle"
+          color="primary"
+          role="status"
+          position="absolute"
+          dangerouslySetInlineStyle={{ __style: { top: 4, right: 4 } }}
+        />
+      ) : null}
```

Comment:
The above is the notification indicator on the top-right of the icon. It is displayed only is collapsed state.

Reply:
![Screenshot by Dropbox Capture](https://github.com/pinterest/gestalt/assets/10593890/677db197-9bb0-4bc5-9852-81815a35debe)